Foundation/NSNumber.swift

```diff
@@ -423,9 +423,17 @@ extension Float : _ObjectTypeBridgeable {
     }
 
     public init?(exactly number: NSNumber) {
-        guard let value = Double(exactly: number) else { return nil }
-        guard let result = Float(exactly: value) else { return nil }
-        self = result
+        let type = number.objCType.pointee
+        if type == 0x49 || type == 0x4c || type == 0x51 {
+            guard let result = Float(exactly: number.uint64Value) else { return nil }
```

Comment:
Can `_cfNumberType()` be used to use names instead of magic numbers?

Reply:
It looks like `_cfNumberType()` doesn't distinguish between signed and unsigned integer types, which could lead to overflow with large unsigned integers.

Reply:
The cf storage always stores signed types so using the base constants are fine

Reply:
With the caveat that you need to up-cast appropriately: but those values are fine as-is

Reply:
How do you handle values that are out of range for `Int64`?

```swift
let x = (UInt64(Int64.max) + 1) as NSNumber
// x.uint64Value == 9223372036854775808
// x.int64Value == -9223372036854775808
```

Reply:
the out of range values are handled via kCFNumberSInt128Type (using CFSInt128Struct)

Reply:
Wow, TIL… I think that would even bitcast to a `DoubleWidth<Int64>`.